services/autoscaling/src/simcore_service_autoscaling/modules/auto_scaling_mode_computational.py

```diff
@@ -0,0 +1,139 @@
+import collections
+import logging
+
+from fastapi import FastAPI
+from models_library.docker import DockerLabelKey
+from models_library.generated_models.docker_rest_api import Node
+from pydantic import AnyUrl, ByteSize
+from servicelib.logging_utils import LogLevelInt
+from servicelib.utils import logged_gather
+
+from ..core.settings import get_application_settings
+from ..models import (
+    AssociatedInstance,
+    DaskTask,
+    EC2InstanceData,
+    EC2InstanceType,
+    Resources,
+)
+from ..utils import computational_scaling as utils
+from ..utils import ec2, utils_docker
+from . import dask
+from .auto_scaling_mode_base import BaseAutoscaling
+from .docker import get_docker_client
+
+_logger = logging.getLogger(__name__)
+
+
+def _scheduler_url(app: FastAPI) -> AnyUrl:
+    app_settings = get_application_settings(app)
+    assert app_settings.AUTOSCALING_DASK  # nosec
+    return app_settings.AUTOSCALING_DASK.DASK_MONITORING_URL
+
+
+class ComputationalAutoscaling(BaseAutoscaling):
+    @staticmethod
+    async def get_monitored_nodes(app: FastAPI) -> list[Node]:
+        return await utils_docker.get_worker_nodes(get_docker_client(app))
+
+    @staticmethod
+    def get_ec2_tags(app: FastAPI) -> dict[str, str]:
+        app_settings = get_application_settings(app)
+        return ec2.get_ec2_tags_computational(app_settings)
+
+    @staticmethod
+    def get_new_node_docker_tags(app: FastAPI) -> dict[DockerLabelKey, str]:
+        assert app  # nosec
```

Comment:
you can also do it like so, not sure if something else complains
```suggestion
        _ = app
```
```suggestion
        assert app  # nosec
```

Reply:
not sure what you are suggesting here @GitHK .